### PR TITLE
Enforce local style source at the end

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/issues/1965

Tweaked logic and dnd to always put local style source in the end and tokens before it.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
- [ ] hi @taylornowotny, I need you to do
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

**Comment by kof** [comment]
- [ ] What should happen to the "backspace" it used to delete the last token, but it can't delete local, so it does nothing

File: apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx [hunk 1]
```diff
@@ -81,10 +94,24 @@ export const useSortable = <Item extends { id: string }>({
...
+
+      return itemId;
     },
-    onStart({ data }) {
-      setDragItemId(data);
+    onStart({ data: itemId }) {
+      const item = items.find((item) => item.id === itemId);
+      if (items.at(-1) === item && item?.source === "local") {
+        toast.error(
+          "Local style source should be last and cannot be reordered"
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:108]: @taylornowotny this wording ok? its shown when you try to reorder local

**Comment by kof** [comment]
We have 2 options with backspace: 
1. delete this ability entirely
2. delete the token after the local

I am not sure 2 is not going to cause problems, because local looks visually like other tokens.

**Comment by taylornowotny** [comment]
I suggest removing the backspace-delete functionality as it no longer makes sense with Local being at the end.

It would not be used in the redesigned Styles Sources UI either, if we ever end up building that.

File: apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx (hunk 1 above)
Comment (taylornowotny) [reply apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:108]: Good idea @TrySound , wording can improve to: "Local token is always last and can not be moved"

**Comment by kof** [comment]
> I suggest removing the backspace-delete functionality as it no longer makes sense with Local being at the end.
> 
> It would not be used in the redesigned Styles Sources UI either, if we ever end up building that.

alright

File: apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx [hunk 2]
```diff
@@ -81,10 +94,22 @@ export const useSortable = <Item extends { id: string }>({
...
+      }
+
+      return itemId;
     },
-    onStart({ data }) {
-      setDragItemId(data);
+    onStart({ data: itemId }) {
+      const item = items.find((item) => item.id === itemId);
+      if (items.at(-1) === item && item?.source === "local") {
+        toast.error("Local token is always last and can not be moved");
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:107]: Local style, not token

Comment (taylornowotny) [reply apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:107]: This is the wording that I chose. If you want we can change it to "Local style source" but not "style" by itself

Comment (taylornowotny) [reply apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:107]: or perhaps more semantic: "Local styles are always last and can not be moved."

Comment (kof) [reply apps/builder/app/builder/features/style-panel/style-source/use-sortable.tsx:107]: works, thanks

**Review by kof — APPROVED** [review]

**Pull request merged**